File: client/src/data/entities/event.ts
export interface Event {
  id: string,
  guildId: string,
  hostId: string,
  title: string,
  subtitle: string,
  description: string,
  createdDate: string,
  startDate: string,
  endDate: string,
  entryOpenDate: string,
  entryCloseDate: string,
  type: EventType
}

export interface CreateEventModel {
  guildId: string,
  title: string,
  subtitle: string,
  description: string,
  createDate: string,
  startDate: string,
  endDate: string,
  entryOpenDate: string,
  entryCloseDate: string
  type: EventType
}

export enum EventType {
  TileRace,
  Bingo,
}

File: API/Entities/Event.cs
namespace API.Entities;

public class Event
{
    public required string Id { get; set; }
    public required string Title { get; set; }
    public string? Subtitle { get; set; }
    public string? Description { get; set; }

    public required string GuildId { get; set; }
    public Guild? Guild { get; set; }

    public required string HostId { get; set; }
    public AppUser? Host { get; set; }

    public bool Started { get; set; }
    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public DateTime? EntryOpenDate { get; set; }
    public DateTime? EntryCloseDate { get; set; }

    public required EventType Type { get; set; }
    
    public enum EventType
    {
        TileRace,
        Bingo
    }
}